DemoApp/Screens/DemoAppTabBarController.swift

```diff
@@ -69,4 +80,39 @@ class DemoAppTabBarController: UITabBarController, CurrentChatUserControllerDele
         let totalUnreadBadge = unreadCount.channels + unreadCount.threads
         UIApplication.shared.applicationIconBadgeNumber = totalUnreadBadge
     }
+
+    func currentUserControllerDidStartSharingLiveLocation(
+        _ controller: CurrentChatUserController
+    ) {
+        debugPrint("[Location] Started sharing live location.")
+        locationProvider.startMonitoringLocation()
+    }
+
+    func currentUserControllerDidStopSharingLiveLocation(_ controller: CurrentChatUserController) {
+        debugPrint("[Location] Stopped sharing live location.")
+        locationProvider.stopMonitoringLocation()
+    }
+
+    func currentUserController(
+        _ controller: CurrentChatUserController,
+        didChangeActiveLiveLocationMessages messages: [ChatMessage]
+    ) {
+        guard !messages.isEmpty else {
+            return
+        }
+
+        messages.forEach {
+            LocationAttachmentSnapshotView.snapshotsCache.removeObject(forKey: $0.id as NSString)
```

Comment:
This will completely change, I'm changing the implementation to how iMessage works. Not a good time to review this now. I'll let you know once everything is ready 👍